### PR TITLE
Use svg as img src to avoid transparent edges

### PR DESCRIPTION
**Pull request by addieachan**
Addresses issue ampproject/amphtml#18133

### CODE REVIEW

**Comment by addieachan** [comment]
@choumx PTAL

File: optimizer/lib/transformers/AddBlurryImagePlaceholders.js [hunk 1]
```diff
@@ -74,15 +74,27 @@ class AddBlurryImagePlaceholders {
...
-    return this.getDataURI_(img).then(() => {
+    img.attribs.src = src;
+    return this.getDataURI_(img).then((dataURI) => {
+      let svg = '<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://ww'
+      + `w.w3.org/1999/xlink" viewBox="0 0 ${dataURI.width} ${dataURI.height}">`
+      + '<filter id="b" color-interpolation-filters="sRGB"><feGaussianBlur stdD'
+      + 'eviation=".5"></feGaussianBlur><feComponentTransfer><feFuncA type="dis'
+      + 'crete" tableValues="1 1%0A"></feFuncA></feComponentTransfer></filter><'
+      + 'image filter="url(#b)" x="0" y="0" height="100%" width="100%" xlink:hr'
+      + `ef="${dataURI.src}"></image></svg>`;
```
Comment (choumx) [inline optimizer/lib/transformers/AddBlurryImagePlaceholders.js]: Can we avoid line breaks in the middle of an attribute and add line breaks after children?

```js
const svgOpen = '<svg xmlns="http://www.w3.org/2000/svg" ' 
    + 'xmlns:xlink="http://www.w3.org/1999/xlink"' 
    + `viewBox="0 0 ${dataURI.width} ${dataURI.height}">`;
const filter = '<filter id="b" color-interpolation-filters="sRGB">';
const blur = '<feGaussianBlur stdDeviation=".5"></feGaussianBlur>';

const html = svgOpen + filter + blur + ...;
```

File: optimizer/lib/transformers/AddBlurryImagePlaceholders.js [hunk 2]
```diff
@@ -74,15 +74,27 @@ class AddBlurryImagePlaceholders {
...
+      svg = svg.replace(/"/g, '\'');
+      svg = encodeURI(svg);
+      svg = svg.replace(/%20/g, ' ');
```
Comment (choumx) [inline optimizer/lib/transformers/AddBlurryImagePlaceholders.js]: Add a comment explaining why?

File: optimizer/lib/transformers/AddBlurryImagePlaceholders.js [hunk 3]
```diff
@@ -74,15 +74,36 @@ class AddBlurryImagePlaceholders {
...
   addBlurryPlaceholder_(tree, src) {
     const img = tree.createElement('img');
-    img.attribs.src = src;
-    img.attribs.class = 'i-amphtml-blur';
+    img.attribs.class = 'i-amphtml-blurry-placeholder';
     img.attribs.placeholder = '';
-    return this.getDataURI_(img).then(() => {
+    img.attribs.src = src;
+    return this.getDataURI_(img).then((dataURI) => {
+      const svgOpen = '<svg xmlns="http://www.w3.org/2000/svg" '
```
Comment (sebastianbenz) [inline optimizer/lib/transformers/AddBlurryImagePlaceholders.js]: nit: for readability I'd prefer this to be a  properly formatted multi-line string literal. 

```
`<svg xmlns="http://www.w3.org/2000/svg" 
       xmlns:xlink="http://www.w3.org/1999/xlink>
    <image filter="url(#b)" x="0" y="0" ...
`
```

You can update `svg.replace(/%20/g, ' ');` to remove multiple whitespaces resulting from the formatting.

**Comment by choumx** [comment]
@sebastianbenz I think this is ready for merge. 

[Travis is failing](https://travis-ci.org/ampproject/amp-toolbox/jobs/434164701) due to an unrelated error on the `runtime-version` project because we currently have a canary version that does *not* start with `00` (031809251804390).

**Comment by choumx** [comment]
@sebastianbenz Ok to merge?

**Comment by sebastianbenz** [comment]
Thanks!

**Pull request merged**